### PR TITLE
New parent job runner for cached data

### PR DESCRIPTION
**Pull request by AndreaFrancis**
Currently, we have datasets_based_job_runner but we need a new one that only creates a chance folder without modifying the datasets library config.
Context https://github.com/huggingface/datasets-server/pull/1296#discussion_r1232512427

### CODE REVIEW

File: services/worker/tests/job_runners/test__datasets_based_worker.py
Comment (severo) [inline services/worker/tests/job_runners/test__datasets_based_worker.py:1]: the new file should also have its tests file

File: services/worker/src/worker/job_runners/_cached_based_job_runner.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+from libcommon.processing_graph import ProcessingStep
+from libcommon.storage import init_dir, remove_dir
+from libcommon.utils import JobInfo
+
+from worker.config import AppConfig
+from worker.job_runner import JobRunner
+
+
+class CachedBasedJobRunner(JobRunner):
+    """Base class for job runners that use a cached directory."""
```
Comment (severo) [inline services/worker/src/worker/job_runners/_cached_based_job_runner.py:20]:
```suggestion
    """Base class for job runners that use a temporary cache directory."""
```

File: services/worker/src/worker/job_runners/_cached_based_job_runner.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+from libcommon.processing_graph import ProcessingStep
+from libcommon.storage import init_dir, remove_dir
+from libcommon.utils import JobInfo
+
+from worker.config import AppConfig
+from worker.job_runner import JobRunner
+
+
+class CachedBasedJobRunner(JobRunner):
```
Comment (severo) [inline services/worker/src/worker/job_runners/_cached_based_job_runner.py:19]:
```suggestion
class CacheDirectoryJobRunner(JobRunner):
```

File: services/worker/src/worker/job_runners/_cached_based_job_runner.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        super().__init__(
+            job_info=job_info,
+            app_config=app_config,
+            processing_step=processing_step,
+        )
+        self.base_cache_directory = cache_directory
+
+    def get_cache_subdirectory(self, digits: int = 14) -> str:
+        random_str = f"{random.randrange(10**(digits - 1), 10**digits)}"  # nosec B311
+        # TODO: Refactor, need a way to generate payload based only on provided params
```
Comment (severo) [inline services/worker/src/worker/job_runners/_cached_based_job_runner.py:41]: do we need this comment? what is the idea?

File: services/worker/src/worker/job_runners/_datasets_based_job_runner.py [hunk 4]
```diff
@@ -10,10 +10,10 @@
 from libcommon.utils import JobInfo
 
 from worker.config import AppConfig
-from worker.job_runners._cached_based_job_runner import CachedBasedJobRunner
+from worker.job_runners._cached_directory_job_runner import CacheDirectoryJobRunner
```
Comment (severo) [inline services/worker/src/worker/job_runners/_datasets_based_job_runner.py:13]:
```suggestion
from worker.job_runners._cache_directory_job_runner import CacheDirectoryJobRunner
```

File: services/worker/src/worker/job_runners/_cached_based_job_runner.py (hunk 3 above)
Comment (AndreaFrancis) [reply services/worker/src/worker/job_runners/_cached_based_job_runner.py]: I copied it from the previous implementation, currently, it generates the payload from job_info (dataset, config, and split) but I thought that in the future we could have a better way to generate it dynamically. 
But I can remove it does not makes sense.

File: services/worker/tests/job_runners/test__datasets_based_worker.py
Comment (AndreaFrancis) [reply services/worker/tests/job_runners/test__datasets_based_worker.py:1]: Done, I separated the current test file in two according to the scenario.

**Review by polinaeterna — APPROVED** [review]
awesome! i will also use it in https://github.com/huggingface/datasets-server/pull/1199

**Pull request merged**